Sources/_FoundationInternals/String/String+Internals.swift

```diff
@@ -16,4 +16,32 @@ extension String {
             $0.properties.isWhitespace
         })
     }
+
+    static func _tryFromUTF16(_ input: UnsafeBufferPointer<UInt16>) -> String? {
+        // Allocate input.count * 3 code points since one UTF16 code point may require up to three UTF8 code points when transcoded
+        withUnsafeTemporaryAllocation(of: UInt8.self, capacity: input.count * 3) { contents in
+            var ptr = contents.baseAddress!
+            var count = 0
+            let error = transcode(input.makeIterator(), from: UTF16.self, to: UTF8.self, stoppingOnError: true) { codeUnit in
+                ptr.pointee = codeUnit
+                ptr = ptr.advanced(by: 1)
+                count += 1
+            }
```

Comment:
Since `transcode()` has no awareness of the size of the temporary allocation, I would suggest adding a debug-mode check that `count` is less than `contents.count`. I think it would be useful for our future selves.